### PR TITLE
fix use_full_image==false case

### PR DESCRIPTION
**Pull request by jonsneyers**
Some fixes to the case where the full modular image is skipped:

- don't assume that everything happens at the modular AC group level (`minShift==0`), there can also be upsampling causing channels to have nonzero shift even when there's no Squeeze
- for partial decodes (when `zerofill` is true), don't try to fill the full image when it's not used. Instead initialize the decoded image with zeroes and skip the decoding.

### CODE REVIEW

File: lib/jxl/dec_modular.cc [hunk 1]
```diff
@@ -274,44 +274,35 @@ Status ModularFrameDecoder::DecodeGroup(const Rect& rect, BitReader* reader,
...
   }
+  if (zerofill && use_full_image) return true;
   ModularOptions options;
-  if (!ModularGenericDecompress(
-          reader, gi, /*header=*/nullptr, stream.ID(frame_dim), &options,
-          /*undo_transforms=*/-1, &tree, &code, &context_map))
+  if (!zerofill && !ModularGenericDecompress(reader, gi, /*header=*/nullptr,
+                                             stream.ID(frame_dim), &options,
+                                             /*undo_transforms=*/-1, &tree,
+                                             &code, &context_map))
```
Comment (veluca93) [inline lib/jxl/dec_modular.cc:295]: Braces pls

Comment (jonsneyers) [reply lib/jxl/dec_modular.cc]: Updated. Also changed the `&&` to two ifs, to make it a bit clearer that short-circuit evaluation is important.

**Review by veluca93 — APPROVED** [review]

**Comment by deymo** [comment]
@jonsneyers  please put the PR description in the commit message, you can do so manually when doing "Squash and merge" (which also tags the commit with the PR number).

**Comment by deymo** [comment]
The code this is fixing was introduced in #205 which is not in v0.5.x branch, removing label. (`use_full_image` is not present in v0.5.x)

**Pull request merged**